File: src/app/ports/in/EstudianteServicePort.php
<?php

namespace src\app\ports\in;

use src\domain\entities\Estudiante;

interface EstudianteServicePort
{
    public function registrarEstudiante(Estudiante $estudiante): void;
    
    public function obtenerEstudiantePorId(int $id): ?Estudiante;
    
    public function listarEstudiante(): array;
    
    public function actualizarEstudiante(Estudiante $estudiante): void;
    
    public function eliminarEstudiante(int $id): void;

}

File: backend-estudiantes/src/main/java/com/jcaa/hexagonal/app/ports/in/EstudianteServicePort.java
package com.jcaa.hexagonal.app.ports.in;

import com.jcaa.hexagonal.domain.entities.Estudiante;
import java.util.List;
import java.util.Optional;

public interface EstudianteServicePort {

    Estudiante crear(Estudiante estudiante);

    Optional<Estudiante> obtenerPorId(Integer id);

    List<Estudiante> listar();

    Estudiante actualizar(Estudiante estudiante);

    void eliminar(Integer id);
}
